doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+  {
+    // Update the text in RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Joint Space Goal", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    // BEGIN_SUB_TUTORIAL joint_space
+    // Planning to a joint-space goal
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // One of the easiest ways to move the robot is by adjusting the joint angles. To start, we'll
+    // create an pointer that references the current robot's state. RobotState is the object that
+    // contains all the current position/velocity/acceleration data.
+    moveit::core::RobotStatePtr current_state = move_group.getCurrentState();
+    //
+    // Next get the current set of joint values for the group.
+    std::vector<double> joint_group_positions;
+    current_state->copyJointGroupPositions(joint_model_group, joint_group_positions);
+
+    // Now, let's modify one of the joints, plan to the new joint space goal, and visualize the
+    // plan.
+    joint_group_positions[0] = -1.0;  // radians
+    move_group.setJointValueTarget(joint_group_positions);
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    // END_SUB_TUTORIAL
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 1 (joint space goal) %s", success ? "" : "FAILED");
+
+    // Visualize the plan in RViz
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' in the window to move to the joint space goal");
+  }
+
+  void plan_pose_goal()
+  {
+    // BEGIN_SUB_TUTORIAL pose_goal
+    // Planning to a Pose goal
+    // ^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // Instead of changing the joint angles a Pose goal chooses a final location and orientation
+    // to end at. The pose contains a quaternion orientation and an xyz location. Here we set
+    // the geometry_msgs::Pose target_pose to our destination and set it to the target.
+    target_pose.orientation.w = 0.0;
+    target_pose.orientation.x = 0.383;
+    target_pose.orientation.y = -0.9240;
+    target_pose.orientation.z = -0.0;
+    target_pose.position.x = 0.28;
+    target_pose.position.y = -0.2;
+    target_pose.position.z = 0.5;
+
+    move_group.setPoseTarget(target_pose);
+    // Setting the plan with a Pose goal is identical to planning with a joint_space goal ::
+    //
+    //     bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    //
+    // END_SUB_TUTORIAL
+
+    // Visualizing the target location axis in RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Pose Goal", rvt::WHITE, rvt::XLARGE);
+    visual_tools.publishAxisLabeled(target_pose, "pose");
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' in the window to plan to the set Pose");
+
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
```

Comment:
this should be in the sub-tutorial

Reply:
It kind of is. The strange comments a few lines up puts an identical line into the tutorial but I avoided the actual line because I didn't want the redundant visual_tools_ lines in the tutorial. I figured a strange comment a few lines up was better than a 1 line sub-tutorial.